### DISCUSSION TITLE
Unable to resolve outdated conversations and merge

### DISCUSSION THREAD

**Discussion by TonyArra** [community/community#19206] [Pull Requests]
Is anyone else having this issue over the last week or two?

We have the option enabled to block merging a PR when there are unresolved conversations. Recently, if a conversation becomes outdated due to a new commit, I am unable to resolve the conversation unless I was the one to make the initial comment. 

The creator of the PR and other members of the team are no longer able to see the conversation, but the PR is prevented from merging due to there being "unresolved" conversations

**Comment by pablomusumeci** [comment]
I'm experiencing the same issue. @TonyArra Have you found any work-around for unblocking the PR with outdated comments?

Edit: This is a duplicate of https://github.com/github-community/community/discussions/10592. I managed to find and resolve the unresolved conversation by scrolling through the entire PR. So there is a way of unblocking the merge but it's just bad UX from Github.

**Comment by ibober** [comment]
I have the same issue with this PR
https://github.com/KhronosGroup/glTF/pull/1977
It says there are 4 unresolved conversations, but I can't find them, really, how I should do this?
Also before some of them disappeared, I noticed that there were conversations that I couldn't resolve - no resolve button.
![image](https://user-images.githubusercontent.com/8260211/184005956-552a931d-f24d-414c-b5a1-d18833bed685.png)

**Reply by aquanow-eric** [reply]
I am getting the same issue. Any workarounds?

**Reply by aquanow-eric** [reply]
Turns out I need to click a button to load all conversations ... once that happens I can get at it and resolve.

**Reply by kopax-polyconseil** [reply]
That's when conversation still exist and have not been overriden by a force push, and then you are stuck!

**Reply by andreieftimie** [reply]
> That's when conversation still exist and have not been overriden by a force push, and then you are stuck!

Any solution when this happens?

**Reply by andreieftimie** [reply]
> > That's when conversation still exist and have not been overriden by a force push, and then you are stuck!
> 
> Any solution when this happens?

Managed to find them in the Conversations tab.

**Reply by xbtmatt** [reply]
Make sure to unhide all hidden items. Couldn't find a conversation because it was lost due to a rebase and the commit disappearing from the repo until I just hit the unhide button for all items (not conversations, although it was a conversation that was hidden...)

**Comment by pmanole-volvocars** [comment]
Same here in 2025.

Clicked View:
<img width="984" alt="image" src="https://github.com/user-attachments/assets/9656786f-a8f2-49ca-982b-324da2727fc4" />


Got here, but cannot find it, and the dropdown doesn't have the caret!
<img width="482" alt="image" src="https://github.com/user-attachments/assets/e519a7aa-f6f5-4f3a-9562-142b0e122f11" />

**Reply by walangitan** [reply]
Managed to find the unresolved conversation by manually finding it in the main pull request body. The link in the conversations tab did not work.

**Comment by bedrich-schindler** [comment]
Problem persists. In this [pull request](https://github.com/react-ui-org/react-ui/pull/544/files), it shows unresolved conversations:

![image](https://github.com/user-attachments/assets/90389830-27ee-4137-b7db-fc07d5d6cc01)
![image](https://github.com/user-attachments/assets/0045450b-75f0-4d29-8157-f29c9aaaff59)

But when I tried to open it, it shows:

![image](https://github.com/user-attachments/assets/a399c05b-6542-41cc-b686-429659d4dbd5)

I also can't find it in the file itself as it was removed with squash/force-push. This results in inability to merge pull request.

**Comment by vyvo2412** [comment]
I'm having this issue as well. Clicking 'Unresolved conversations' does not take me anywhere.
![image](https://github.com/user-attachments/assets/2453d6b5-4100-4e6e-85b9-8c24946f2744)
![image](https://github.com/user-attachments/assets/4a6a7658-38fb-4426-b66b-db09eee65895)

**Comment by ricardobeat** [comment]
Also having this issue right now blocking a PR merge.

In my case, the unresolved comments were on commits that have been removed by a rebase.

**Comment by qweered** [comment]
Same issue, cannot resolve my threads in nixpkgs repo

**Comment by rafaelbeckel** [comment]
I had this issue today, and it turns out there was a HIDDEN conversation in one of the collapsed blocks. If you're facing this, most probably you should painstakingly open them all until you find it.

It's still a bug, though, unresolved conversations should not be collapsible.

**Reply by rafaelbeckel** [reply]
I faced this issue again, this time it's a stale comment over a commit that do not exist anymore after rebase (branch was force-pushed), but GitHub still flags the comment as unresolved and blocks the merge.

**Reply by rafaelbeckel** [reply]
These are the commands that solved it:

```
gh api graphql -f query='query($owner:String!, $repo:String!, $pr:Int!){ repository(owner:$owner, name:$repo) { pullRequest(number:$pr) { reviewThreads(first: 100) { nodes { id isResolved comments(first: 50) { nodes { id url author { login } } } } } } } }' -F owner='YOUR_ORG_NAME' -F repo='YOUR_REPOSITORY' -F pr=80 | jq -r '
  .data.repository.pullRequest.reviewThreads.nodes
  | map(select(.isResolved==false))
  | map(select([.comments.nodes[].author.login] | any(test("copilot"; "i"))))
  | [ .[] | [.id, ([.comments.nodes[].author.login] | unique | join(","))] | @tsv ]
  | if length==0 then "NONE" else .[] end
'
PRRT_kwDOPU1kOs5d0RI2	copilot-pull-request-reviewer <--- the culprit
```

```
owner='YOUR_ORG_NAME'; repo='kyoso-YOUR_REPOSITORY'; pr=80;
ids=$(gh api graphql -f query='query($owner:String!, $repo:String!, $pr:Int!){ repository(owner:$owner, name:$repo) { pullRequest(number:$pr) { reviewThreads(first: 100) { nodes { id isResolved comments(first: 50) { nodes { author { login } } } } } } } }' -F owner="$owner" -F repo="$repo" -F pr=$pr | jq -r '
  .data.repository.pullRequest.reviewThreads.nodes
  | map(select(.isResolved==false))
  | map(select([.comments.nodes[].author.login] | any(test("copilot"; "i"))))
  | .[].id
');
for id in $ids; do
  gh api graphql -f query='mutation($id:ID!){ resolveReviewThread(input:{threadId:$id}) { thread { id isResolved } } }' -f id="$id" >/dev/null && echo "Resolved $id";
done
Resolved PRRT_kwDOPU1kOs5d0RI2 <--- DONE
```

**Comment by rayrayraykk** [comment]
Still exists.

**Comment by pietrofxq** [comment]
Facing the same problem. This one was caused by Copilot review
<img width="732" height="221" alt="image" src="https://github.com/user-attachments/assets/3968d807-5239-4b57-9fbe-523564b220ff" />

**Comment by ninikurshavishvili** [comment]
This appears to happen when review threads are attached to commits that no longer exist (force-push / rebase / squash), but GitHub still keeps the thread in an unresolved state.

A few things that helped in different cases:

UI workarounds (sometimes works):

Open the Conversations tab and use “Unhide all”

Scroll the entire PR timeline (including collapsed / outdated blocks)

Expand every collapsed diff — unresolved threads can live inside them

If the conversation is truly orphaned (no longer visible in UI):

The only reliable fix I’ve seen is resolving it via the GitHub GraphQL API using gh

This seems especially common with Copilot review comments, which can remain unresolved after rebases

Unfortunately, when the original commit is gone, the UI gives no way to resolve the thread — which feels like a GitHub bug rather than user error.

Would be great if GitHub either:

Auto-resolved threads when the target commit disappears, or

Exposed orphaned review threads clearly in the UI instead of blocking merges silently.

**Comment by dhisonp** [comment]
Still happening!

**Comment by sebastiancreatics** [comment]
Does anybody have a solution to this? @github please fix

**Comment by fedricantony** [comment]
Going through commit by commit helps me to resolve this issue. Here is what I did.

- Open the PR
- Go to the Commits tab
- Open each commit and check the diff
- Look for old unresolved comments
- Resolve them

**Reply by fedricantony** [reply]
Or look at the PR overview for outdated commit. Then go to that commit and resolve comments

**Comment by Alameen688** [comment]
<img width="466" height="135" alt="conversation" src="https://github.com/user-attachments/assets/515d3644-57e9-4d5f-a1a3-c4968f98e1b1" />

There is a small conversation icon on the right section of the PR, click on it, and it will display unresolved comments.

**Comment by noromer-fadv** [comment]
It's amazing that this issue has been a problem for over 4 years, and yet Github doesn't care to just make the "can't merge due to unresolved comments" dialog provide a link to the comments in question. It's not complicated, but it does require Github to care about their users.

**Comment by thomasrebele** [comment]
I tried rafaelbeckel's [workaround](https://github.com/orgs/community/discussions/19206#discussioncomment-14690192), but I get an error message `gh: thomasrebele does not have the correct permissions to execute ResolveReviewThread`. Unfortunately I cannot set the permissions as I'm not the owner of the repository (and the required permissions are too wide, see https://github.com/orgs/community/discussions/44650).

I cannot resolve the conversation, even when I'm the original commenter.
